vip-integrations.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * Plugin Name: VIP Integrations
+ * Description: Plugin loading for VIP plugin integrations.
+ * Author: Automattic
+ * License: GPL version 2 or later - http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+ */
+
+namespace Automattic\VIP\Integrations;
+
+defined( 'ABSPATH' ) || die();
+
+require_once __DIR__ . '/vip-integration-helper/integrations/integrations.php';
+require_once __DIR__ . '/vip-integration-helper/integrations/block-data-api.php';
+
+add_action( 'muplugins_loaded', __NAMESPACE__ . '\\register_integrations', /* priority */ 4 );
+
+/**
+ * Register valid integrations.
+ */
+function register_integrations(): void {
+	Integrations::instance()->register( 'block-data-api', BlockDataApi::class );
+}
+
+/**
+ * Activates an integration with an optional configuration value.
+ *
+ * @param string $slug   A unique identifier for the integration.
+ * @param array  $config An associative array of configuration values for the integration.
+ */
+function activate_integration( $integration_slug, $config = [] ) {
+	add_action( 'muplugins_loaded', function() use ( $integration_slug, $config ) {
+		$integration = Integrations::instance()->get_registered( $integration_slug );
+
+		if ( null !== $integration ) {
+			$integration->activate( $config );
+		}
+	}, /* priority */ 6 );
+}
```

Comment:
So it's clear in my mind...

Currently:
 - `muplugins_loaded:4` - built-in integrations are registered.
 - `muplugins_loaded:6` - registered integrations activated via a call in `vip-config.php` happen at this point.
 - `muplugins_loaded:8` - active integrations have their `integrate()` method called, which, at least for block-data-api allows it to hook into `plugins_loaded` and do its thing.
 
That only gives someone a gap of `muplugins_loaded:<6` to register or unregister their Integration (any later and activate_integrations will be hooking into muplugins_loaded:6 when the new Integration hasn't been registered yet).

Reply:
That's correct - there's a bit of space (`4`, `6`, `8`) to allow injecting between the setup priorities, but it's not designed for that explicitly. Each VIP `Integration` should be registered by us via `register_vip_integrations()` which will ensure activation occurs later. As long as our VIP integrations are registered there, the priorities can be ignored by integration implementations.

The priorities are using sub-`10` values so that integrations can register with the default `muplugins_loaded` (`10`) if desired.